apps/bot/src/fish.ts

```diff
@@ -0,0 +1,90 @@
+import { Hashira } from "@hashira/core";
+import { randomInt } from "es-toolkit";
+import { base } from "../base";
+import { STRATA_CZASU_CURRENCY } from "../specializedConstants";
+import { addBalance } from "./managers/transferManager";
+import { formatBalance } from "./util";
+import type { ExtendedPrismaClient } from "@hashira/db";
+import { isAfter, addMinutes } from "date-fns";
+
+const checkIfRedeemable = async (
+  prisma: ExtendedPrismaClient,
+  userId: string,
+  guildId: string,
+) => {
+  const redeems = await prisma.lastFishing.findMany({
+    where: { userId, guildId },
+    orderBy: { timestamp: "desc" },
+  });
+
+    const lastRedeem = redeems.length > 0 ? redeems[0].timestamp : new Date(0);

```

Comment:
`redeems[0]` -> error `Object is possibly 'undefined'.ts(2532)`
```suggestion
    const lastRedeem = redeems[0]?.timestamp ?? new Date(0);
```